### PR TITLE
V2 : Upgrade JSON Serializer

### PR DESCRIPTION
**Pull request by z3dev**
This pull request updates the JSON serializer to use V2. The goals of this pull request are:
- [X] Update to use V2 MODELING API
- [X] Update code up to JSCAD coding standards
- [X] Add test suites to complete coverage
  * 100% Statements
  * 80% Branches
  * 100% Functions
  * 100% Lines

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Does your submission pass tests?

### CODE REVIEW

**Comment by z3dev** [comment]
@kaosat-dev this is ready for review now. all objects are serialized to JSON, reflecting the same in deserialize.

File: packages/io/io/formats.js [hunk 1]
```diff
@@ -50,6 +51,14 @@ const formats = {
     deserializable: true,
     convertGeom3: true,
     convertGeom2: true },
+  json: {
```
Comment (kaosat-dev) [inline packages/io/io/formats.js:54]: this is good addition :+1:

File: packages/io/json-serializer/README.md [hunk 2]
```diff
@@ -1,15 +1,19 @@
 ## @jscad/json-serializer
 
-> json serializer for the jscad project (from CSG)
+> Serializer of JSCAD geometries to JSON notation
```
Comment (kaosat-dev) [inline packages/io/json-serializer/README.md:3]: JSON notation is redundant simply JSON is fine :)

File: packages/io/json-serializer/README.md [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
 ## Overview
 
-This serializer outputs a 'blobable' array of data (from a CSG object)
-ie an array that can either be passed directly to a Blob (`new Blob(blobable)`)
-or converted to a Node.js buffer.
+This serializer outputs a 'blobable' array of data from one or more JSCAD geometries.
+The array of data can either be used to create a Blob (`new Blob(blobable)`), or converted to a Node.js buffer.
+
+The serialization of the following geometries are possible.
+- serialization of 2D geometry (geom3) to JSON notation (string)
```
Comment (kaosat-dev) [inline packages/io/json-serializer/README.md:14]: same

File: packages/io/json-serializer/index.js [hunk 4]
```diff
@@ -1,95 +1,66 @@
...
-    throw new Error('no arguments supplied to serialize function !')
-  } else if (params.length === 1) {
-    // assumed to be object(s)
-    objects = Array.isArray(params[0]) ? params[0] : params
-  } else if (params.length > 1) {
-    options = params[0]
-    objects = params[1]
+// Replace all typed arrays in geometries with standard Arrays
+// NOTE: 'this' in replacer is the object in which key was found
+function replacer (key, value) {
```
Comment (kaosat-dev) [inline packages/io/json-serializer/index.js:23]: haha awesome !

File: packages/io/json-serializer/README.md (hunk 2 above)
Comment (z3dev) [reply packages/io/json-serializer/README.md]: Done

File: packages/io/json-serializer/README.md (hunk 3 above)
Comment (z3dev) [reply packages/io/json-serializer/README.md]: Done

**Comment by z3dev** [comment]
@kaosat-dev thanks again for the excellent review. this is ready to go now. :)

**Pull request merged**